plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2551,8 +2551,7 @@ else if (isMetadataColumnId(columnHandle.getId())) {
 
         if (newEnforcedConstraint.equals(table.getEnforcedPredicate())
                 && newUnenforcedConstraint.equals(table.getUnenforcedPredicate())
-                && newConstraintColumns.equals(table.getConstraintColumns())
-                && constraint.getPredicateColumns().isEmpty()) {
```

Comment:
@marton-bod / @zhengxingmao - What was the initial reason for adding a check on `constraint.getPredicateColumns().isEmpty()` - Ideally the `applyFilter` should depends on the `IcebergTableHandle` state after applying the filter and not based on `Constraint`'s state